pkg/protoc/providers.go

```diff
@@ -0,0 +1,65 @@
+package protoc
+
+import (
+	"sync"
+
+	"github.com/google/go-cloud/wire"
+	"github.com/izumin5210/gex"
+	"github.com/izumin5210/gex/pkg/tool"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"k8s.io/utils/exec"
+
+	"github.com/izumin5210/grapi/pkg/cli"
+)
+
+var (
+	gexCfg   *gex.Config
+	gexCfgMu sync.Mutex
+
+	toolRepo   tool.Repository
+	toolRepoMu sync.Mutex
+)
+
+func ProvideGexConfig(
+	fs afero.Fs,
+	execer exec.Interface,
+	io *cli.IO,
+	rootDir cli.RootDir,
+) *gex.Config {
+	gexCfgMu.Lock()
+	defer gexCfgMu.Unlock()
+	if gexCfg == nil {
+		gexCfg = &gex.Config{
+			OutWriter:  io.Out,
+			ErrWriter:  io.Err,
+			InReader:   io.In,
+			FS:         fs,
+			Execer:     execer,
+			WorkingDir: rootDir.String(),
+			// TODO: set verbose flag
+			// TODO: set logger
+		}
+	}
+	return gexCfg
+}
+
+func ProvideToolRepository(gexCfg *gex.Config) (tool.Repository, error) {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function ProvideToolRepository should have comment or be unexported